mesonbuild/backend/xcodebackend.py

```diff
@@ -585,9 +585,15 @@ def generate_pbx_aggregate_target(self, objects_dict: PbxDict) -> None:
                 elif isinstance(d, build.BuildTarget):
                     dependencies.append(self.pbx_dep_map[d.get_id()])
             for s in t.sources:
-                if not isinstance(s, build.GeneratedList):
+                if isinstance(s, build.GeneratedList):
+                    build_phases.append(self.shell_targets[(tname, generator_id)])
+                elif isinstance(s, build.ExtractedObjects):
+                    source_target_id = self.pbx_dep_map[s.target.get_id()]
+                    if source_target_id not in dependencies:
+                        dependencies.append(source_target_id)
+                    continue
+                else:
                     continue
-                build_phases.append(self.shell_targets[(tname, generator_id)])
                 for d in s.depends:
                     dependencies.append(self.pbx_custom_dep_map[d.get_id()])
                 generator_id += 1
```

Comment:
It makes more sense to me to drop the double continue and fold the GeneratedList only code into that block